File: app/src/main/java/com/thewizrd/simpleweather/snackbar/SnackbarManagerInterface.java
package com.thewizrd.simpleweather.snackbar;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface SnackbarManagerInterface {
    @Nullable
    SnackbarManager createSnackManager();

    void initSnackManager();

    void showSnackbar(@NonNull Snackbar snackbar, @Nullable com.google.android.material.snackbar.Snackbar.Callback callback);

    void dismissAllSnackbars();

    void unloadSnackManager();
}

File: app/src/main/java/com/thewizrd/simpleweather/snackbar/SnackbarManagerInterface.kt
package com.thewizrd.simpleweather.snackbar

import android.app.Activity
import com.google.android.material.snackbar.Snackbar as MaterialSnackbar

interface SnackbarManagerInterface {
    fun createSnackManager(activity: Activity): SnackbarManager?
    fun initSnackManager(activity: Activity)
    fun showSnackbar(snackbar: Snackbar, callback: MaterialSnackbar.Callback?)

    fun dismissAllSnackbars()
    fun unloadSnackManager()
}